canu/generate/switch/config/config.py

```diff
@@ -565,7 +570,8 @@ def generate_switch_config(
     if switch_name not in sls_variables["HMN_IPs"].keys():
         click.secho(f"Cannot find {switch_name} in CSI / SLS nodes.", fg="red")
         exit(1)
-
+    if sls_variables["CMN_IPs"]:
```

Comment:
The way Python works checking this w/o it existing will blow up.  You could:
```
if "CMN_IPs" in sls_variables:
```
But a more Pythonic way to do this is:
```
myvar = sls_variables.get("CMN_IPs")
```
which returns None if the variable isn't set and you can test on that or...
```
variables["CMN_IP"] = sls_variables.get("CMN_IPs", whatever_you_want_default_to_be)
```

Reply:
Fixed.